doc/pages/architecture/060-triggers.md

```diff
@@ -1,23 +1,374 @@
+
+`device_trigger_type` can be one of the following:
+- `device_connected`: triggered when a device connects to the MQTT broker.
+- `device_disconnected`: triggered when a device disconnects from the MQTT broker.
+
+`device_id` can be a specific Device ID or `*`, meaning the trigger will be installed for all
+devices in a Realm.
+
+### Data Triggers
+
+Data triggers express conditions matching data coming from a device.
+
+This is the generic representation of a Device Trigger:
+
+```json
+{
+  "type": "data_trigger",
+  "on": "<data_trigger_type>",
+  "interface_name": "<interface_name>",
+  "interface_major": "<interface_major>",
+  "match_path": "<match_path>",
+  "value_match_operator": "<value_match_operator>",
+  "known_value": <known_value>
+}
+```
+
+Data triggers are installed for all devices in a Realm.
+
+#### Parameters
+
+`data_trigger_type` can be one of the following:
+- `incoming_data`: verifies the condition whenever new data arrives.
+- `value_stored`: verifies the condition whenever new data arrives, after it is saved to the
+  database.
+- `value_change`: verifies the condition whenever a new sample is not equal to the last one.
+- `value_change_applied`: verifies the condition whenever a new sample is not equal to the last one,
+  after it is saved to the database.
+- `path_created`: verifies the condition whenever a new property interface is set or the first value
+  is streamed on a datastream interface.
+- `path_created`: verifies the condition whenever a property is unset.
+
+`interface_name` and `interface_major` represent, respectively, the Interface name and major version
+that uniquely identify an Astarte Interface. `interface_name` can be `*` to match all interfaces; in
+that case `interface_major` is ignored.
+
+`match_path` is the path that will be used to match the condition. It can be `/*` to match all the
+paths of an interface.
+
+`value_match_operator` is the operator used to match the incoming data against a known value. It can
+be `*` to indicate that all values should be matched (`known_value` is ignored in that case),
+otherwise it can be one of these operators: `==`, `!=`, `>`, `>=`, `<`, `<=`, `contains`,
+`not_contains`. The match is always performed with `<incoming_value> <operator> <known_value>`.
+
+`known_value` is the value used with `value_match_operator` to perform the comparison on the
+incoming value. It must have the same type as the incoming value.
+
+## Actions
+
+Actions are triggered by a matching condition. An Action defines how the event should be sent to the
+outer world (e.g. an http POST on a certain URL). In addition, most actions have a Payload, which
+carries the body of the event.
+
+Payloads are most of the time represented as text, and Astarte provides several ways to generate
+them. By default Astarte generates a JSON payload with all the relevant information of the event.
+This is also the format used when delivering payloads in Astarte Channels. The format of the payload
```

Comment:
Cross-link Astarte Channels documentation here.

Reply:
The one in the User Guide? Currently there's no Channels documentation in the Architecture part besides the one at the bottom of this page